pallets/msa/src/lib.rs

```diff
@@ -1068,6 +1146,40 @@ impl<T: Config> Pallet<T> {
 		})
 	}
 
+	/// Adds an association between MSA id and ProviderRegistryEntry. As of now, the
+	/// only piece of metadata we are recording is provider name.
+	///
+	/// # Events
+	/// * [`Event::ProviderCreated`]
+	///
+	/// # Errors
+	/// * [`Error::NoKeyExists`] - account does not have an MSA
+	/// * [`Error::ExceedsMaxProviderNameSize`] - Too long of a provider name
+	/// * [`Error::DuplicateProviderRegistryEntry`] - a ProviderRegistryEntry associated with the given MSA id already exists.
+	///
+	pub fn do_create_provider(
```

Comment:
Maybe we can simplify this function to something like this and use this in the extrinsic.
```
pub fn create_registered_provider(
		provider_id: ProviderId,
		name: BoundedVec<u8, T::MaxProviderNameSize>,
	) -> DispatchResult {
		ProviderToRegistryEntry::<T>::try_mutate(provider_id, |maybe_metadata| -> DispatchResult {
			ensure!(maybe_metadata.take().is_none(), Error::<T>::DuplicateProviderRegistryEntry);
			*maybe_metadata = Some(ProviderRegistryEntry { provider_name: name });
			Ok(())
		})
	}
```

Reply:
This has to stay here.

Reply:
can we please rename this to create_provider